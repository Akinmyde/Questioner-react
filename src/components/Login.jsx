import React from 'react';
import { Link, Redirect } from 'react-router-dom';
import Form from './common/Form';
import Loader from './common/Loader';
import loginSchema from '../schema/loginSchema';
import { login, getCurrentUser } from './services/authService';
import exceptionHandler from '../helpers/exceptionHandler';

class Login extends Form {
  constructor(props) {
    super(props);
    this.state = { data: { email: '', password: '' }, loading: false };
  }

  schema = { ...loginSchema };

  doSubmit = async () => {
    const { email, password } = this.state.data;
    this.setState({ loading: true });
    try {
      await login({ username: email, password });
      const { location } = this.props;
      const { state } = location;
      const path = state ? state.from.pathname : '/';
      this.props.history.push(path);
    } catch (ex) {
      exceptionHandler(ex);
    } finally {
      this.setState({ loading: false });
    }
  };

  render() {
    if (getCurrentUser()) return <Redirect to="/" />;

    const { loading } = this.state;
    return (
      <React.Fragment>
        {loading && <Loader />}
        <div className="container">
          <div className="imgcontainer">
            <i className="fas fa-user-circle fa-5x" />
          </div>
          {this.renderInput('email', 'Email', 'JohnDoe@gmail.com', 'email')}
          {this.renderInput('password', 'Password', 'password', 'password')}
          <div className="right font12"><Link to="/forget">Forget Password</Link></div>
          {this.renderButton('Login')}
          <div className="center font12">
            {'Don\'t have an account?'}
            {' '}
            <Link to="/signup">Sign up</Link>
          </div>
          <div className="error font12" />
        </div>
      </React.Fragment>
    );
  }
}

export default Login;

