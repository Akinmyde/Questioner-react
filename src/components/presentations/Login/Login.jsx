import React from 'react';
import { Link, Redirect } from 'react-router-dom';
import Form from '../../common/Form';
import Loader from '../../common/Loader';
import loginSchema from '../../../schema/loginSchema';

class Login extends Form {
  constructor(props) {
    super(props);
    this.state = { data: { email: '', password: '' } };
  }

  schema = { ...loginSchema };

  doSubmit = async () => {
    const { login } = this.props;
    const { email, password } = this.state.data;
    await login({ username: email, password }, this.props);
  };

  render() {
    const { auth, LoadingReducer } = this.props;
    const { userId } = auth;
    const { loader } = LoadingReducer;

    if (userId) return <Redirect to="/" />;
    return (
      <React.Fragment>
        {loader && <Loader />}
        <div className="container">
          <div className="imgcontainer">
            <i className="fas fa-user-circle fa-5x" />
          </div>
          {this.renderInput('email', 'Email', 'JohnDoe@gmail.com', 'email')}
          {this.renderInput('password', 'Password', 'password', 'password')}
          <div className="right font12"><Link to="/forget">Forget Password</Link></div>
          {this.renderButton('Login')}
          <div className="center font12">
            {'Don\'t have an account?'}
            {' '}
            <Link to="/signup">Sign up</Link>
          </div>
          <div className="error font12" />
        </div>
      </React.Fragment>
    );
  }
}

export default Login;

