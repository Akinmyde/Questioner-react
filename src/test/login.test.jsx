import 'babel-polyfill';
import React from 'react';
import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer'
import Login from '../components/Login';

describe('Login Page', () => {
  it('should render login correctly', () => {
    const app = shallow(<Login />);
    expect(app).toMatchSnapshot();
  });

  it('should change state', () => {
    const wrapper = shallow(
      <Login />
    );
    expect(wrapper.state('data')).toEqual({ email: '', password: '' });
  });

  it('should test functions', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.instance().doSubmit()).toMatchSnapshot();
  });
});
