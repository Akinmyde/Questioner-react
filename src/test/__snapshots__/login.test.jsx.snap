// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render login correctly 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="imgcontainer"
    >
      <i
        className="fas fa-user-circle fa-5x"
      />
    </div>
    <Input
      autoFocus=""
      id="email"
      label="Email"
      onChange={[Function]}
      placeholder="JohnDoe@gmail.com"
      type="email"
      value=""
    />
    <Input
      autoFocus=""
      id="password"
      label="Password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <div
      className="right font12"
    >
      <Link
        to="/forget"
      >
        Forget Password
      </Link>
    </div>
    <Button
      className="btn"
      onClick={[Function]}
      value="Login"
    />
    <div
      className="center font12"
    >
      Don't have an account?
       
      <Link
        to="/signup"
      >
        Sign up
      </Link>
    </div>
    <div
      className="error font12"
    />
  </div>
</Fragment>
`;

exports[`Login Page should test functions 1`] = `Promise {}`;
